akismet-privacy-policies.php

```diff
@@ -38,33 +46,49 @@ class Akismet_Privacy_Policies {
 	 * @return \Akismet_Privacy_Policies
 	 */
 	public function __construct() {
-
-		// The plugin is only helpful on german blogs
-		if ( 'de_DE' !== get_locale() ) {
-			return;
-		}
-
 		register_deactivation_hook( __FILE__, array( &$this, 'unregister_settings' ) );
 		register_uninstall_hook( __FILE__, array( 'Akismet_Privacy_Policies', 'unregister_settings' ) );
 
 		add_filter( 'comment_form_defaults', array( $this, 'add_comment_notice' ), 11, 1 );
 		add_action( 'akismet_privacy_policies', array( $this, 'add_comment_notice' ) );
 
-		$options = get_option( 'akismet_privacy_notice_settings' );
-		if ( empty( $options[ 'checkbox' ] ) ) {
-			$options[ 'checkbox' ] = $this->checkbox;
+		$this->languages = get_available_languages();
+		// default language is en_US but get_available_languages
+		// contains translations in .mo files
+		$this->languages[] = 'en_US';
+
+		if ( isset( $_GET[ 'translation' ])) {
+			$this->translation = $_GET[ 'translation' ];
+		} elseif( isset( $_POST[ 'translation' ])) {
+			$this->translation = $_POST[ 'translation' ];
+		} else {
+			$this->translation = get_user_locale();
+		}
+
+		if ( $this->translation != 'en_US' ) {
```

Comment:
Please check always strict for clear coding, speed --> `!==`. However in this context, it is also a idea to leave `en_US` for other language keys wit `en`? So like `en_ NZ` have also the chance to use this translation content?

Reply:
How could this be handled efficiently? Are there special versions of WP for English speaking countries other than US? I have chosen `en_US` as I assumed this to be the default language and all other versions to be translations through `.po` files. Should the language test possibly be handled through a regular expression to filter out all possible versions of English?